src/inmanta/module.py

```diff
@@ -2342,3 +2572,92 @@ def get_all_python_requirements_as_list(self) -> List[str]:
+            new_install_requires = [
+                r
+                for r in config_parser.get("options", "install_requires").split("\n")
+                if r and Requirement.parse(r).key != python_pkg_requirement.key
+            ]
+            new_install_requires.append(str(python_pkg_requirement))
+        else:
+            new_install_requires = [str(python_pkg_requirement)]
+        config_parser.set("options", "install_requires", "\n".join(new_install_requires))
+        # Write config back to disk
+        with open(self.get_metadata_file_path(), "w", encoding="utf-8") as fd:
+            config_parser.write(fd)
+        # Reload in-memory state
+        with open(self.get_metadata_file_path(), "r", encoding="utf-8") as fd:
+            self._metadata = ModuleV2Metadata.parse(fd)
+
+    def has_module_requirement(self, module_name: str) -> bool:
+        config_parser = ConfigParser()
+        config_parser.read(self.get_metadata_file_path())
+        if not config_parser.has_option("options", "install_requires"):
+            return False
+        pkg_name: str = ModuleV2.get_package_name_for(module_name)
+        return any(
+            r
+            for r in config_parser.get("options", "install_requires").split("\n")
+            if r and Requirement.parse(r).key == pkg_name
+        )
+
+
+class YamlMetadataFileWithRequiresField:
+    @classmethod
+    def add_module_requirement_and_write(
+        cls,
+        proj_or_v1_mod: Union[Project, ModuleV1],
+        requirement: InmantaModuleRequirement,
+        ignore_version_constraint: bool = False,
+    ) -> None:
+        """
+        Updates the metadata file of the given project or module by adding the given requirement the `requires` section.
+
+        :param proj_or_v1_mod: Project or module that should receive the new requirement.
+        :param requirement: The requirement to add.
+        :param ignore_version_constraint: If True, the version constraint in `requirement` will be left out of the .yml file.
+        """
+        # Parse cfg file
+        content: CommentedMap = PreservativeYamlParser.parse(proj_or_v1_mod.get_metadata_file_path())
+        line_to_add = str(requirement) if not ignore_version_constraint else requirement.key
+        # Update requires
+        if "requires" in content:
+            existing_matching_reqs: List[str] = [
+                r for r in content["requires"] if InmantaModuleRequirement.parse(r).key == requirement.key
+            ]
+            for r in existing_matching_reqs:
+                content["requires"].remove(r)
+            content["requires"].append(line_to_add)
+        else:
+            content["requires"] = [line_to_add]
+        # Write file back to disk
+        PreservativeYamlParser.dump(proj_or_v1_mod.get_metadata_file_path(), content)
+
+    @classmethod
+    def has_module_requirement(cls, proj_or_v1_mod: Union[Project, ModuleV1], module_name: str) -> bool:
```

Comment:
Here as well I believe it would be good to state that `module_name` should be lower case.